rollup.config.mjs

```diff
@@ -32,6 +32,15 @@ const plugins = [
  * @param {*} warn
  */
 function onwarn(warning, warn) {
+  // we error at any warning.
+  // we allow-list the errors we understand are not harmful
+  if (
+    warning.code === 'PLUGIN_WARNING' &&
+    !warning.message.includes('sourcemap')
+  ) {
+    console.error(chalk.redBright(warning.message));
+    throw Object.assign(new Error(), warning);
+  }
```

Comment:
@ShaMan123 let's have the build error on default if types are broken.
We can comment this locally when we need to work in a broken state with watchers, but i fill better if the pipeline is strict about this, otherwise we found ourselves with unexpected cleanups that we can't see when we code review the PRs

Reply:
I agree
That sourcemap warning must be resoled as well